homeassistant/components/squeezebox/config_flow.py

```diff
@@ -0,0 +1,256 @@
+                vol.Optional(CONF_PASSWORD): str,
+            }
+        )
+
+    async def async_step_user(self, user_input=None, errors=None):
+        """Handle a flow initialized by the user."""
+        if user_input and CONF_HOST in user_input:
+            # update with host provided by user
+            self.data_schema = vol.Schema(
+                {
+                    vol.Required(
+                        CONF_HOST,
+                        description={"suggested_value": user_input.get(CONF_HOST)},
+                    ): str,
+                    vol.Required(CONF_PORT, default=DEFAULT_PORT,): int,
+                    vol.Optional(CONF_USERNAME): str,
+                    vol.Optional(CONF_PASSWORD): str,
+                }
+            )
+            return await self.async_step_edit()
+
+        # no host specified, see if we can discover an unconfigured LMS server
+        try:
+            await asyncio.wait_for(self._discover(), timeout=TIMEOUT)
+            return await self.async_step_edit()
+        except asyncio.TimeoutError:
+            return self.async_show_form(
+                step_id="user",
+                data_schema=vol.Schema({vol.Optional(CONF_HOST): str}),
+                errors={"base": "no_server_found"},
+            )
+
+        # display the form
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema({vol.Optional(CONF_HOST): str}),
+            errors=errors,
+        )
+
+    async def async_step_edit(self, user_input=None):
+        """Edit a discovered or manually inputted server."""
+        errors = {}
+        if user_input:
+            try:
+                info = await validate_input(self.hass, user_input)
+                if "uuid" in info:
+                    await self.async_set_unique_id(info["uuid"])
+                    self._abort_if_unique_id_configured()
+                return self.async_create_entry(title=info.get("ip"), data=user_input)
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+            except InvalidAuth:
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                errors["base"] = "unknown"
+
+        return self.async_show_form(
+            step_id="edit", data_schema=self.data_schema, errors=errors
+        )
+
+    async def async_step_import(self, config, errors=None):
+        """Import a config flow from configuration."""
+        try:
+            DATA_SCHEMA(config)
+            info = await validate_input(self.hass, config)
+            if "uuid" in info:
+                try:
+                    await self.async_set_unique_id(info["uuid"])
+                    # update with info from configuration.yaml
+                    self._abort_if_unique_id_configured(info)
+                except data_entry_flow.AbortFlow as error:
+                    return self.async_abort(reason=error.reason)
```

Comment:
```suggestion
                await self.async_set_unique_id(info["uuid"])
                # update with info from configuration.yaml
                self._abort_if_unique_id_configured(info)
```

If you don't trap the exception it will abort on its own